src/Pos/Lrc/Arbitrary.hs

```diff
@@ -0,0 +1,73 @@
+-- | Arbitrary instances for Lrc types.
+
+module Pos.Lrc.Arbitrary
+       ( InvalidRichmenStake (..)
+       , ValidRichmenStake (..)
+       ) where
+
+import           Universum
+
+import qualified Data.HashMap.Strict as HM
+import           Pos.Constants       (genesisMpcThd)
+import           Pos.Lrc.Types       (RichmenStake)
+import           Pos.Types.Address   (StakeholderId)
+import           Pos.Types.Core      (Coin, coinPortionDenominator, getCoinPortion,
+                                      mkCoin, unsafeGetCoin)
+import           Test.QuickCheck     (Arbitrary (..), Gen, choose)
+
+-- | Wrapper over 'RichmenStake'. Its 'Arbitrary' instance enforces that the stake
+-- distribution inside must be valid, i.e. all of the coins are non-zero and every
+-- stakeholder has sufficient coins to participate.
```

Comment:
@gromakovsky What would be the merits of 3? I haven't given that much thought, so if there are very good reasons I'll start doing that when possible.